### PR TITLE
Added decoding for encoding of ordered column profiles

### PR DESCRIPTION
**Pull request by ksneab7**

### CODE REVIEW

File: dataprofiler/profilers/order_column_profile.py [hunk 1]
```diff
@@ -270,6 +270,22 @@ def report(self, remove_disabled_flag: bool = False) -> dict:
...
+        :param data: dictionary with attributes and values.
+        :type data: dict[string, Any]
+
+        :return: Profiler with attributes populated.
+        :rtype: CategoricalColumn
+        """
+        # This is an ambiguous call to super classes.
+        profile = super().load_from_dict(data)
+
+        return profile
```
Comment (micdavis) [inline dataprofiler/profilers/order_column_profile.py:287]:
```suggestion
        return super().load_from_dict(data)
```
I see benefits to either way, just putting another option out there

File: dataprofiler/tests/profilers/test_order_column_profile.py [hunk 2]
```diff
@@ -413,3 +415,41 @@ def test_json_encode_after_update(self):
...
+        expected_profile = OrderColumn(fake_profile_name)
+
+        serialized = json.dumps(expected_profile, cls=ProfileEncoder)
+        deserialized = load_column_profile(json.loads(serialized))
+
+        utils.assert_profiles_equal(deserialized, expected_profile)
+
+    def test_json_decode_after_update(self):
+        fake_profile_name = "Fake profile name"
+        # Actual deserialization
```
Comment (micdavis) [inline dataprofiler/tests/profilers/test_order_column_profile.py:430]:
```suggestion
```
dont think this comment is needed

File: dataprofiler/profilers/order_column_profile.py (hunk 1 above)
Comment (taylorfturner) [reply dataprofiler/profilers/order_column_profile.py:287]: cleaner +1

Comment (taylorfturner) [reply dataprofiler/profilers/order_column_profile.py:287]: @ksneab7

Comment (taylorfturner) [reply dataprofiler/profilers/order_column_profile.py]: the other implementation do operations on `profile`, but in this case we don't .... so no need I don't think to set it to a variable

**Review by micdavis — APPROVED** [review]

**Review by taylorfturner — APPROVED** [review]
LGTM

File: dataprofiler/tests/profilers/test_order_column_profile.py [hunk 3]
```diff
@@ -413,3 +415,40 @@ def test_json_encode_after_update(self):
...
+        serialized = json.dumps(expected_profile, cls=ProfileEncoder)
+        deserialized = load_column_profile(json.loads(serialized))
+
+        utils.assert_profiles_equal(deserialized, expected_profile)
+
+    def test_json_decode_after_update(self):
+        fake_profile_name = "Fake profile name"
+
+        # Build expected orderColumn
+        df_order = pd.Series(["za", "z", "c", "a"])
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/test_order_column_profile.py:432]: might have be interesting to have this ordered and then the update below keep it ordered and validating the output expectation order. Maybe can quick add.

Comment (ksneab7) [reply dataprofiler/tests/profilers/test_order_column_profile.py:432]: I see what hes saying there `order` attribute is not consistent after updating

**Pull request merged**